### PR TITLE
Feature: Refactor Sudoku Solver Project Boilerplate for a Backend Focus to Improve Testability

### PR DESCRIPTION
**Pull request by SaintPeter**
As discussed here:
https://github.com/freeCodeCamp/freeCodeCamp/issues/39692

@nhcarrigan and I refactored this project to have a more backend focus to improve testability by the fCC test infrastructure.

Key Changes:
* Removed all front-end related User Stories 
* Rewrote the front-end to provide minimal features, treating it as a basic API endpoint test
* Added a new requirement to check the validity of an individual value's placement
* Rewrote both the unit tests to cover the algorithm in more depth
  * Added specific tests for checking placement for row/column/region
* Rewrote functional tests to cover the two new endpoints.

NOTE: This change should not be merged until the (as yet uncreated) `/learn` PR has been completed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,2 @@
...
-1. All 4 functional tests are complete and passing. See `/tests/2_functional-tests.js` for the functionality you should write tests for.
-
-### Testing and additional notes
-
-* To run the tests on Repl.it, set NODE_ENV to test without quotes in the .env file.
-* To run the tests in the console, use the command npm run test. To open the Repl.it console, press Ctrl+Shift+P (Cmd if on a Mac) and type "open shell".
-* All logic can go into `public/sudoku-solver.js`.
-* Create all of the unit/functional tests in `tests/1_unit-tests.js` and `tests/2_functional-tests.js`.
+# **freeCodeCamp** 
+### [Quality Assurance Projects - Sudoku Solver](https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-projects/sudoku-solver)
```
Comment (ShaunSHamilton) [inline README.md]:
```suggestion
# [Sudoku Solver](https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-projects/sudoku-solver)
```

**Review by ShaunSHamilton — DISMISSED** [review]
Some clarification about these changes:
- We have decided to remove the block name from the headings, to prevent the case where the project is moved, and then need to update the boilerplate READMEs

Otherwise, looking good, Rex

File: views/index.html [hunk 2]
```diff
@@ -1,926 +1,309 @@
...
+	<meta charset="utf-8" />
+	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
+	<meta name="viewport" content="width=device-width, initial-scale=1" />
+	<link rel="stylesheet" href="../public/style.css" type="text/css" />
+</head>
+
+<body>
+	<header>
+		<h1>
+			Quality Assurance 4: Sudoku Solver
```
Comment (ShaunSHamilton) [inline views/index.html]:
```suggestion
			Sudoku Solver
```

File: controllers/sudoku-solver.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class SudokuSolver {
+  //if string length is < 81, return "You too short man"
```
Comment (ShaunSHamilton) [inline controllers/sudoku-solver.js]: Does this serve any purpose?

Comment (naomi-lgbt) [reply controllers/sudoku-solver.js]: The comment? Nope. I think that's left over from our whiteboarding process 😅

Comment (ShaunSHamilton) [reply controllers/sudoku-solver.js]: Right. Rex, could you remove it, then.

Thanks, for all of this. You two have done excellent work

File: views/index.html (hunk 2 above)
Comment (SaintPeter) [reply views/index.html]: This is resolved in the latest commit.

**Comment by SaintPeter** [comment]
This should be good to go now.

**Review by moT01 — APPROVED** [review]
LGTM 🎉 

My testing process was to run all three PR's for this project locally, submit the demo project on the website - then move the code from there into the few files on this one and submit that - it went smooth and I think I am good with this.

**Comment by RandellDawson** [comment]
@Sky020 If you are ok with the changes, I am too, so you can approve and then we can just wait for the `/learn` PR changes to hit production once that PR is merged.

**Review by moT01 — APPROVED** [review]
LGTM 🎉 - The PR to main repo is now on production so we should merge this to prevent confusion. Thanks again for all the work on these @SaintPeter @nhcarrigan 🎉

**Pull request merged**